console/client/src/hooks/use-schema.ts

```diff
@@ -0,0 +1,66 @@
+import { Code, ConnectError } from '@connectrpc/connect'
+import { useEffect, useState } from 'react'
+import { useClient } from '../hooks/use-client'
+import { useVisibility } from '../hooks/use-visibility.ts'
+import { ControllerService } from '../protos/xyz/block/ftl/v1/ftl_connect.ts'
+import { DeploymentChangeType, PullSchemaResponse } from '../protos/xyz/block/ftl/v1/ftl_pb'
+
+export const useSchema = () => {
+  const client = useClient(ControllerService)
+  const [schema, setSchema] = useState<PullSchemaResponse[]>([])
+  const isVisible = useVisibility()
+
+  useEffect(() => {
+    const abortController = new AbortController()
+
+    const fetchSchema = async () => {
+      try {
+        if (!isVisible) {
+          abortController.abort()
+          return
+        }
+
+        const schemaMap = new Map<string, PullSchemaResponse>()
+        for await (const response of client.pullSchema(
+          {},
+          {
+            signal: abortController.signal,
+          },
+        )) {
```

Comment:
This is the key change here on this PR. For some reason, `pullSchema` is perfectly happy with `signal` being passed in the first argument 🤷